addons/addon-base-raas/packages/base-raas-workflow-steps/lib/steps/provision-account/provision-account.js

```diff
@@ -96,22 +97,102 @@ class ProvisionAccount extends StepBase {
     this.print(`saving account data into dynamo: ${JSON.stringify(data)}`);
     await accountService.saveAccountToDb(requestContext, data, accountId);
     // THIS IS NEEDED, we should wait AWS to setup the account, even if we can fetch the account ID
-    this.print('start to wait for 5min for AWS getting the account ready.');
+    this.print('start to wait for 5 minutes for AWS getting the account ready.');
+
+    if (this.settings.get(settingKeys.isAppStreamEnabled) === 'true') {
+      return this.wait(60 * 5).thenCall('shareImageWithMemberAccount');
+    }
     return this.wait(60 * 5).thenCall('deployStack');
   }
 
+  async shareImageWithMemberAccount() {
+    const [accountService] = await this.mustFindServices(['accountService']);
+    const requestContext = await this.payload.object('requestContext');
+    const accountId = await this.state.string('ACCOUNT_ID');
+    const appStreamImageName = await this.payload.string('appStreamImageName');
+    await accountService.shareAppStreamImageWithMemberAccount(requestContext, accountId, appStreamImageName);
+
+    return this.wait(10).thenCall('createAppStreamRoles');
+  }
+
+  async createAppStreamRoles() {
```

Comment:
Can you add some unit tests for the newly added code in this file. We should also add tests around isAppStream flag and verify the expected behavior.

Reply:
Let's merge this PR for now to speed up development for other devs. Unit tests can follow in a separate PR.